app/src/main/java/bean/pwr/imskamieskiego/MapDrawer/MapDrawer.java

```diff
@@ -233,7 +244,40 @@ private Bitmap layerPathAndTacks(Collection<MapPoint> mapObjects, List<MapPoint>
                 }
             }
         }
-        return floorToDrawOn;
+        return mutableBitmap;
+    }
+
+    private Bitmap convertBitmapToMutable(Bitmap bitmap){
```

Comment:
Kod tej metody powstał w oparciu o ten temat na stackoverflow
https://stackoverflow.com/questions/4349075/bitmapfactory-decoderesource-returns-a-mutable-bitmap-in-android-2-2-and-an-immu/9194259#9194259
Przyznam szczerze, nie podoba mi się ten kod. Wpis na stacku jest sprzed 8 lat i rozwiązuje problemy wstecznej kompatybilności z androidem w wersji 1.6. Obawiam się, że ten kod jest już mocno przestarzały. Poza tym, zapisywanie bitmapy do externalStorage wydaje mi się takim niezbyt fajnym pomysłem. Wymaga to dodatkowych uprawnień itd. Jeśli już, czy nie lepszy byłby internalStorage?

Reply:
Jak sprawdzałem, to dzięki użyciu tej metody convertBtimapToMutable(), zmniejszała się ilość użytej pamięci RAM jakoś koło 40%. Możliwe, że to rozwiązanie jest zupełnie bez sensu, spróbowałem to jakoś załatać, żeby działało i nie pobierało aż tyle pamięci. Może rzeczywiście lepszą opcją by było aby ten problem został rozwiązany z nową wersja MapDrawera?

Reply:
Myślę, że zrobimy tak: ja to troszkę po testuję, jak to jest z tą zajętością pamięci i potem to wciągniemy jakie jest. Nie mniej jednak jest to coś, co koniecznie musimy mieć na uwadze podczas tworzenia nowej wersji MapDrawera. @KamilSwiech - rzuć okiem na to review, bo to dotyczy Twoich aktualnych zadań